incubator/hnc/internal/reconcilers/object.go

```diff
@@ -650,20 +650,12 @@ func (r *ObjectReconciler) syncPropagation(ctx context.Context, log logr.Logger,
 // - Objects have a selector that doesn't match the destination namespace
 // - Service Account token secrets
 func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstructured.Unstructured, dst string) bool {
-	selector, errSelector := selectors.GetSelector(inst)
-	// TODO: generate events for the errors
-	if errSelector != nil {
-		log.Error(errSelector, "Cannot propagate")
-	}
-	treeSelector, errTreeSelector := selectors.GetTreeSelector(inst)
-	if errTreeSelector != nil {
-		log.Error(errTreeSelector, "Cannot propagate")
-	}
-	noneSelector, errNoneSelector := selectors.GetNoneSelector(inst)
-	if errNoneSelector != nil {
-		log.Error(errNoneSelector, "Cannot propagate")
-	}
 	nsLabels := r.Forest.Get(dst).GetLabels()
+	isMatch, err := selectors.IsSelectorsMatch(inst, nsLabels)
+	if err != nil {
+		// TODO: generate events for the errors
+		log.Error(err, "Cannot propagate")
+	}
```

Comment:
Sorry, remind me why we don't early-return here on error? A comment would be nice.

Honestly I'm not really sure about this whole switch statement anymore. When it started, it looked like there really would be a number of independent cases, but now it just looks like a bunch of `if` statements with early returns. I wonder if that would be more clear.

Reply:
You are right that it can be an early return. We didn't do it before because we want to log multiple errors, but now this can be done in the `DoSelectorsMatch()`. However, I didn't do it because last time we discussed that these two options both make sense, and logging multiple errors in the function seems like an overkill